x-pack/plugins/fleet/server/services/epm/elasticsearch/template/install.ts

```diff
@@ -365,8 +365,9 @@ export function buildComponentTemplates(params: {
           mapping: {
             ...templateSettings.index?.mapping,
             total_fields: {
-              ...templateSettings.index?.mapping?.total_fields,
-              limit: '10000',
+              limit: templateSettings.index?.mapping?.total_fields?.limit
+                ? templateSettings.index?.mapping?.total_fields?.limit
+                : 1000,
```

Comment:
Maybe omit the limit in this case so that we rely on the default in ES?

Reply:
IMHO we should be opinionated here to give integration authors a clear baseline. If they can't rely on a number of fields that will "just work", integrations could break because they were built with the 1k limit in mind and then it suddenly changes in Elasticsearch (or even worse, can be different based on the context).

We have this problem right now by lowering from 10k to 1k: https://github.com/elastic/integrations/pull/9342

Reply:
For these reasons, it would be a breaking change if the default field limit in ES changed, so I don't think that's going to happen. But it probably doesn't matter that much whether we rely on the default in ES or hard-code a default value in Fleet that matches the default in Elasticsearch.